SchedulerBot/SchedulerBot.Business.Entities/ExecutionResult/ExecutionResult.cs

```diff
@@ -0,0 +1,138 @@
+﻿namespace SchedulerBot.Business.Entities
+{
+	/// <summary>
+	/// Describes the result of the operation execution.
+	/// </summary>
+	public class ExecutionResult
+	{
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ExecutionResult"/> class.
+		/// </summary>
+		protected ExecutionResult()
+		{
+			ErrorMessage = null;
+			ErrorCode = ExecutionErrorCodes.None;
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ExecutionResult{T}"/> class.
+		/// </summary>
+		/// <param name="errorCode">The error code.</param>
+		/// <param name="errorErrorMessage">The error message.</param>
+		private ExecutionResult(ExecutionErrorCodes errorCode, string errorErrorMessage)
+		{
+			ErrorMessage = errorErrorMessage;
+			ErrorCode = errorCode;
+		}
+
+		/// <summary>
+		/// Gets a value indicating whether the execution is successful.
+		/// </summary>
+		public bool IsSuccess => ErrorCode == ExecutionErrorCodes.None;
+
+		/// <summary>
+		/// Gets the error message related with the execution result.
+		/// </summary>
+		public string ErrorMessage { get; protected set; }
```

Comment:
I would rather make the setters `private` and set the values from the ctor calls.

Reply:
done